guppylang/compiler/expr_compiler.py

```diff
@@ -554,12 +556,17 @@ def visit_ResultExpr(self, node: ResultExpr) -> Wire:
                 value_wire = self.builder.add_op(read_bool(), value_wire)
             op_name = f"result_{base_name}"
             hugr_ty = base_ty
-        op = tket2_result_op(
-            op_name=op_name,
-            typ=hugr_ty,
-            tag=node.tag,
-            extra_args=extra_args,
+
+        sig = ht.FunctionType(
+            input=[hugr_ty],
+            output=[],
         )
+        args = [
+            ht.StringArg(node.tag),
+            *extra_args,
+        ]
```

Comment:
ultra nit: We could clean up the formatting a bit as a drive-by

```suggestion
        sig = ht.FunctionType(input=[hugr_ty], output=[])
        args = [ht.StringArg(node.tag), *extra_args]
```